tests/unit/h_vialib/secure/expiry_test.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime, timedelta, timezone
+
+import pytest
+
+from h_vialib.secure.expiry import YEAR_ZERO, as_expires, quantized_expiry
+
+
+class TestQuantizedExpiry:
+    @pytest.mark.parametrize("max_age", (99, timedelta(seconds=99)))
+    def test_it_is_quantized(self, max_age):
+        expires = quantized_expiry(max_age=max_age, divisions=3)
+
+        assert isinstance(expires, datetime)
+        diff = expires - YEAR_ZERO
+        assert diff.total_seconds() % (99 / 3) == 0
+
+    def test_it_is_in_the_range_we_expect(self):
+        now = datetime.now(tz=timezone.utc)
+        expires = quantized_expiry(max_age=120, divisions=4)
+
+        diff = expires - now
+
+        # It's in the future
+        assert diff > timedelta(seconds=0)
+        # It's at least max_age * divisions - 1 / divisions
+        assert diff >= timedelta(seconds=120 * (3 / 4))
+        # It's at most max_age
+        assert diff <= timedelta(seconds=120)
+
+    @pytest.mark.parametrize("max_age", (None, "foo"))
+    def test_it_raises_with_invalid_max_age(self, max_age):
+        with pytest.raises(ValueError):
+            quantized_expiry(max_age=max_age)
```

Comment:
Can we test that repeatedly calling it with the same arguments returns the same result? And then after an amount  of time has passed, it starts returning a different result? Similar to the tests I wrote here: https://github.com/hypothesis/via3/blob/c18e328fd946cf5411de893fc05d1b00d33857ac/tests/unit/via/views/view_pdf_test.py#L28-L69

Reply:
I think the conditions here actually are sufficient to prove the behavior in a mathematical sense, but I get that they don't show it intuitively. I'll add some tests which are easier to grok